homeassistant/components/tts/__init__.py

```diff
@@ -490,6 +481,45 @@ def async_get_tts_audio(self, message, language, options=None):
             ft.partial(self.get_tts_audio, message, language, options=options))
 
 
+class TextToSpeechUrlView(HomeAssistantView):
+    """TTS view to get a url to a generated speech file."""
+
+    requires_auth = False
```

Comment:
If we expose the `cache` attribute in the post body, an attacker could use this HTTP view to create quite a bit of load on the server without auth, I think. With a single request with a long text message the server would go and download the TTS audio file which would be much larger in size than the input. Of course requiring auth would make this a bit harder to use too...

Reply:
I think that makes sense. Let me try again with auth.